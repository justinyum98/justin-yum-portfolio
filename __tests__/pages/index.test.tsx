import * as React from 'react';
import { render } from '@testing-library/react';
import Home from '../../pages/index';

describe('Home page', () => {
  it('renders correctly', () => {
    // Arrange
    const { container } = render(<Home />);

    // Act
    
    // Assert
    expect(container).toMatchSnapshot();
  });
});
