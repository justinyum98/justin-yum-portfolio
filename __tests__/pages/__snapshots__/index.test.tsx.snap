// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<div>
  <div
    class="bg-fwTan"
    id="home-page"
  >
    <div
      class="bg-fwTan text-fwOrange sticky top-0 w-full "
    >
      <nav
        aria-label="Header navigation bar"
        class="w-full flex flex-row justify-between items-center px-4 py-2 lg:px-12 lg:py-4 lg:gap-x-4"
        id="header-navbar"
        role="navigation"
      >
        <a
          class="pt-2 px-2 rounded-full hover:bg-white"
          href="/"
        >
          <img
            alt="(Logo)"
            height="48"
            src="/logo.svg"
            width="48"
          />
        </a>
        <button
          class="p-2 lg:hidden"
          id="header-navbar-open-sidebar-button"
        >
          <svg
            fill="currentColor"
            height="48"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="48"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 6H20V8H4zM8 11H20V13H8zM13 16H20V18H13z"
            />
          </svg>
        </button>
        <div
          class="hidden font-sans lg:flex lg:flex-auto lg:gap-x-12 lg:ml-6"
          id="header-navbar-navlinks"
        >
          <a
            aria-label="About page link"
            class="lg:px-2 lg:py-1 lg:text-base hover:text-fwRed"
            href="/#about"
            id="header-navbar-navlinks-about"
            role="link"
          >
            About
          </a>
          <a
            aria-label="Work page link"
            class="lg:px-2 lg:py-1 lg:text-base hover:text-fwRed"
            href="/#work"
            id="header-navbar-navlinks-work"
            role="link"
          >
            Work
          </a>
          <a
            aria-label="Projects page link"
            class="lg:px-2 lg:py-1 lg:text-base hover:text-fwRed"
            href="/#projects"
            id="header-navbar-navlinks-projects"
            role="link"
          >
            Projects
          </a>
          <a
            aria-label="Contact page link"
            class="lg:px-2 lg:py-1 lg:text-base hover:text-fwRed"
            href="/#contact"
            id="header-navbar-navlinks-contact"
            role="link"
          >
            Contact
          </a>
        </div>
        <div
          class="hidden lg:flex lg:flex-auto lg:flex-row-reverse lg:gap-x-12"
          id="header-navbar-socials"
        >
          <a
            aria-label="Linkedin social link"
            class="p-1"
            href="https://www.linkedin.com/in/justinyum98"
            id="header-navbar-socials-linkedin"
            role="link"
          >
            <svg
              fill="currentColor"
              height="28"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1 1 68.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
              />
            </svg>
          </a>
          <a
            aria-label="Github social link"
            class="p-1"
            href="https://github.com/justinyum98"
            id="header-navbar-socials-github"
            role="link"
          >
            <svg
              fill="currentColor"
              height="28"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </div>
      </nav>
    </div>
    <div
      id="home-page-content"
    >
      <div
        class="container mx-auto py-12 px-8 lg:py-24 lg:px-24"
        id="home-page-content-intro"
      >
        <div
          class="flex flex-col font-sans"
        >
          <h3
            class="text-fwOrange text-lg lg:text-xl"
          >
            Hi, my name is
          </h3>
          <div
            class="text-fwDarkPurple mt-1 mb-5 lg:mt-3 lg:mb-7"
          >
            <h1
              class="font-bold text-4xl pb-1 lg:text-7xl lg:pb-3"
            >
              Justin Yum.
            </h1>
            <h1
              class="font-semibold text-3xl pt-1 lg:text-5xl lg:pt-3"
            >
              Software Engineer.
            </h1>
          </div>
          <h2
            class="text-fwDarkPurple text-sm lg:text-base"
          >
            I develop (and sometimes design) both web and mobile experiences.
            <br />
            Currently, I am a
             
            <span
              class="font-medium text-fwOrange"
            >
              junior undergraduate student
            </span>
             
            working part-time at
             
            <span
              class="font-medium text-fwOrange"
            >
              ASML
            </span>
            .
          </h2>
        </div>
      </div>
    </div>
  </div>
</div>
`;
